### PR TITLE
Send compressed LayoutDescription and VersionedPreset

### PR DESCRIPTION
**Pull request by henriquegemignani**
Presets can be rather large (looking at you, listing a lot of starting locations), so the main speed up is not sending the presets.

### CODE REVIEW

File: randovania/gui/lib/multiplayer_session_api.py [hunk 1]
```diff
@@ -244,15 +254,15 @@ async def create_new_world(self, name: str, preset: VersionedPreset, owner: int)
         self.logger.info("Creating world named '%s' with %s for %d", name, preset.name, owner)
         await self._session_admin_player(
             owner, admin_actions.SessionAdminUserAction.CREATE_WORLD_FOR,
-            (name, preset.as_json)
+            (name, preset.as_bytes())
```
Comment (ThanatosGit) [inline randovania/gui/lib/multiplayer_session_api.py:257]: This (and the other occurences) do not work. It returns `bytes` which are not JSON serializable for the server call.

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: randovania/gui/lib/multiplayer_session_api.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/gui/lib/multiplayer_session_api.py:257]: It's perfectly fine to send bytes to the server! In fact even faster

Comment (ThanatosGit) [reply randovania/gui/lib/multiplayer_session_api.py:257]:
```
Traceback (most recent call last):
  File "..\rando-repos\randovania\venv\Lib\site-packages\qasync\__init__.py", line 783, in _error_handler
    task.result()
  File "..\rando-repos\randovania\randovania\gui\widgets\multiplayer_session_users_widget.py", line 243, in _new_world
    await self._session_api.create_new_world(new_name, preset, user_id)
  File "..\rando-repos\randovania\randovania\gui\lib\multiplayer_session_api.py", line 33, in wrapper
    return await fn(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "..\rando-repos\randovania\randovania\gui\lib\multiplayer_session_api.py", line 255, in create_new_world
    await self._session_admin_player(
  File "..\rando-repos\randovania\randovania\gui\lib\multiplayer_session_api.py", line 131, in _session_admin_player
    return await self.network_client.server_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "..\rando-repos\randovania\randovania\network_client\network_client.py", line 422, in server_call
    result = await self.sio.call(event, data, namespace=namespace, timeout=timeout)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "..\rando-repos\randovania\venv\Lib\site-packages\socketio\asyncio_client.py", line 292, in call
    await self.emit(event, data=data, namespace=namespace,
  File "..\rando-repos\randovania\venv\Lib\site-packages\socketio\asyncio_client.py", line 228, in emit
    await self._send_packet(self.packet_class(
  File "..\rando-repos\randovania\venv\Lib\site-packages\socketio\asyncio_client.py", line 352, in _send_packet
    encoded_packet = pkt.encode()
                     ^^^^^^^^^^^^
  File "..\rando-repos\randovania\venv\Lib\site-packages\socketio\packet.py", line 64, in encode
    encoded_packet += self.json.dumps(data, separators=(',', ':'))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "..\AppData\Local\Programs\Python\Python311\Lib\json\__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "..\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "..\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "..\AppData\Local\Programs\Python\Python311\Lib\json\encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

Comment (henriquegemignani) [reply randovania/gui/lib/multiplayer_session_api.py:257]: guess it needs to be the entire message and not part of it

Comment (henriquegemignani) [reply randovania/gui/lib/multiplayer_session_api.py:257]: Ah no. It doesn't automatically detect because I'm using _tuples_. ugh

**Review by ThanatosGit — APPROVED** [review]
Besides the failing tests, looks fine to me.

**Pull request merged**